src/lib/protocols/bittorrent.c

```diff
@@ -69,24 +69,48 @@ static int search_bittorrent_again(struct ndpi_detection_module_struct *ndpi_str
 
 /* *********************************************** */
 
+static int get_utpv1_length(const u_int8_t *payload, u_int payload_len)
+{
+  struct ndpi_utp_hdr *h = (struct ndpi_utp_hdr*)payload;
+  unsigned int off;
+  u_int8_t ext_type = h->next_extension;
+
+  off = sizeof(struct ndpi_utp_hdr);
+  while(ext_type != 0 && off + 1 < payload_len) {
```

Comment:
If I am not mistaken `libutp` has a hard limit of maximum allowed extensions. Was it 4? Well, I can not even remember stuff I read 2 days ago. :D

Reply:
2 days is a looong time :)
I didn't find anything in `libutp` but putting an hard limit makes a lot of sense

Reply:
From https://github.com/bittorrent/libutp/blob/master/utp_internal.cpp on line 510:

```C
	// extension bytes from SYN packet
	byte extensions[8];
```

Reply:
This filed is used to save the *paylaod* of the extension type 2, only. see around line 1857. Am I wrong?

Reply:
Yea, no hard limit for sure. :D